main/main.cpp

```diff
@@ -2406,6 +2406,13 @@ Error Main::setup2() {
 	initialize_navigation_server();
 	register_server_singletons();
 
+#if defined(MODULE_MONO_ENABLED) && defined(TOOLS_ENABLED)
+	// Process command lines relevant to the bindings generator before initializing
+	// the mono module, so it can skip runtime initialization if we're generating glue
+	List<String> cmdline_args = OS::get_singleton()->get_cmdline_args();
+	BindingsGenerator::process_cmdline(cmdline_args);
+#endif
```

Comment:
Why move this here? Can't we keep it below with `BindingsGenerator::generate_bindings()`? I would rather have all the .NET related hacks in one place instead of scattered around the `main.cpp`, if possible.

Reply:
This is the problem - we need to process the command line and set up state before initializing scripting languages. The point where `generate_bindings` gets called happens after the hostfxr initialization, and it will crash if it tries to initialize with an incomplete installation, aborting the binding generation before it has a chance to run. The binding generation doesn't need hostfxr to be initialized, so we have to check the command line and skip the hostfxr initialization if the bindings generator is going to run.

I put this here so it's visually close to where the `BindingsGenerator::generate()` call is, so they're paired up and if someone goes to take one out, they won't miss the other call. I've also put it in `CSharpLanguage::init()` before this, and right before hostfxr initialization before that, and nobody likes any of the places where it goes. I totally get it, this is a hack, so it's not nice to have it anywhere! But we still have to do it in some way, otherwise the runtime is going to crash when trying to generate bindings. The hack was already there, I just had to split it up in order not to crash things while trying to be as minimal and helpful as possible.

Reply:
Ah I see. I'm a little worried that scripting languages initialization may add stuff to ClassDB which the generator would need to know about, but I don't think that's the case.

> nobody likes any of the places where it goes

Well, I personally agree with **neikeq**'s comment (see https://github.com/godotengine/godot/pull/76659#discussion_r1183208369) and, if there's no agreement, I would default to what he recommends since he's the main .NET maintainer.